influxdb3_server/src/http.rs

```diff
@@ -456,6 +469,12 @@ where
     }
 
     async fn authorize_request(&self, req: &mut Request<Body>) -> Result<(), AuthorizationError> {
+        // Extend the request with the authorization token; this is used downstream in some
+        // APIs, such as write, that need the full header value to authorize a request.
+        let auth_header = req.headers().get(AUTHORIZATION).cloned();
```

Comment:
The removal below this code was to make sure we don't log it. Is there some way we can just keep this header, but obfuscate it in logs? We did something like this at my time at Commure where we used the hazy library we made to make a type that wouldn't print out secrets -> https://github.com/commure/hazy/blob/master/examples/simple.rs

Reply:
I think that we are still safe here. `auth_header` here gets wrapped in the `AuthorizationHeaderExtension` which has a [custom `Debug` implementation](https://github.com/influxdata/influxdb3_core/blob/main/authz/src/http.rs#L17-L21) that obscures it in the logs. And then, the header is removed from the `Request` below.

As it happens, we need to put the full value of the authentication header in this extension on the request for the `v1`/`v2` write request parsing logic to work. They both call [this function](https://github.com/influxdata/influxdb3_core/blob/main/iox_http/src/write/single_tenant/auth.rs#L12-L32), which expects it there in that form, or the requests will not authorize.

Reply:
Okay in which case this works for me!